middlewares/team.middleware.js

```diff
@@ -119,6 +120,65 @@ async function ensureSpace(req, res, next) {
     return next();
 }
 
+async function updateTeam(req, res, next) {
+    const team = await Services.Team.updateOne(req.body.team._id, req.body.teamDetails);
+
+    if (!team) {
+        return next({
+            status: 404,
+            message: Constants.Error.TEAM_404_MESSAGE,
+            data: req.body.teamId
+        });
+    }
+
+    req.body.team = team;
+    return next();
+}
+
+async function getByHackerId(req, res, next) {
+    // might be in req.params.hackerId
+    const hacker = await Services.Hacker.findById(req.body.hackerId);
+
+    if (!hacker) {
+        return next({
+            status: 404,
+            message: Constants.Error.HACKER_404_MESSAGE,
+            data: req.body.hackerId
+        });
+    }
+
+    const team = await Services.Team.findById(hacker.teamId);
+
+    if (!team) {
+        return next({
+            status: 404,
+            message: Constants.Error.TEAM_404_MESSAGE,
+            data: hacker.teamId,
+        });
+    }
+
+    req.body.team = team;
+    next();
+}
+
+async function getByUser(req, res, next) {
+    const hacker = await Services.Hacker.findByAccountId(req.user.id);
+
+    if (!hacker) {
+        return next({
+            status: 404,
+            message: Constants.Error.Hacker,
+            data: {
+                id: req.user.id
+            }
+        });
+    }
+
+    // should be cleared by req.logout()
+    req.user.teamId = hacker.teamId;
```

Comment:
I'm not sure whether we want to modify the value of `user`. The alternative is to place it in `body`. What happens if we do that?